synapseclient/table.py

```diff
@@ -473,6 +473,33 @@ def cast_row_set(rowset):
         rowset['rows'][i]['values'] = cast_row(row, rowset['headers'])
     return rowset
 
+
+def _create_row_delete_csv(row_id_vers_iterable):
```

Comment:
Per @zimingd, this could be a context manager that automatically delete the csv file after `deleting` the rows.